server/src/main/java/org/elasticsearch/search/lookup/LeafDocLookup.java

```diff
@@ -50,32 +66,76 @@ public void setDocument(int docId) {
         this.docId = docId;
     }
 
-    protected DocValuesScriptFieldFactory getScriptFieldFactory(String fieldName, MappedFieldType.FielddataOperation options) {
-        DocValuesScriptFieldFactory factory = localCacheScriptFieldData.get(fieldName);
+    /**
+     * @param fieldName the name of the field to retrieve data for
+     * @param isFieldAccess {@code true} if this uses the field-access API, {@code false} if
+     *                                  this uses the doc-access API
+     */
+    protected DocValuesScriptFieldFactory getScriptFieldFactory(String fieldName, boolean isFieldAccess) {
```

Comment:
Would this be better off as two methods, one for field access and one for doc access?

Reply:
It already is as get(String) is used for doc-style access and this method is used for field-style access. Ideally, someday when deprecation occurs all of this can be deleted easily.